src/rgw/rgw_op.cc

```diff
@@ -3758,21 +3754,18 @@ void RGWPutObj::execute(optional_yield y)
     s->dest_placement = *pdest_placement;
     pdest_placement = &s->dest_placement;
     ldpp_dout(this, 20) << "dest_placement for part=" << *pdest_placement << dendl;
-    processor.emplace<MultipartObjectProcessor>(
-        &*aio, store, s->bucket.get(), pdest_placement,
-        s->owner.get_id(), obj_ctx, s->object->clone(),
-        multipart_upload_id, multipart_part_num, multipart_part_str,
-        this, s->yield);
+    processor = upload->get_writer(this, s->yield, s->object->clone(),
+				   s->user->get_id(), obj_ctx, pdest_placement,
+				   multipart_part_num, multipart_part_str);
   } else if(append) {
     if (s->bucket->versioned()) {
       op_ret = -ERR_INVALID_BUCKET_STATE;
       return;
     }
-    pdest_placement = &s->dest_placement;
-    processor.emplace<AppendObjectProcessor>(
-            &*aio, store, s->bucket.get(), pdest_placement, s->bucket_owner.get_id(),
-	    obj_ctx, s->object->clone(),
-            s->req_id, position, &cur_accounted_size, this, s->yield);
+    processor = store->get_append_writer(this, s->yield, s->object->clone(),
```

Comment:
is it necessary to clone a separate unique_ptr here, instead of just sharing a raw pointer/reference? it seems like we can rely on `s->object` to preserve its lifetime

Reply:
There are other places that create a new obj and pass it in, so it's lifetime would end when the function returns.